packages/datadog/src/index.ts

```diff
@@ -12,21 +13,42 @@ export class DataDogMiddleware implements Middleware {
           ...context.payload,
         },
       },
-      () => next()
+      (span: tracer.Span | undefined) => {
+        if (span) {
+          const datadogContextData: Record<string, any> = {};
+          tracer.inject(span, 'text_map', datadogContextData);
+          const meta = this.getMetaFromContext(context);
+          meta.datadog = datadogContextData;
+          context.payload._meta = meta;
+        }
+        next(context);
```

Comment:
This passes the updated contexts to the callback function instead of relying on it to be available globally.